app/controllers/api/projects.php

```diff
@@ -152,37 +125,12 @@
             throw new Exception(Exception::PROJECT_RESERVED_PROJECT, "'console' is a reserved project.");
         }
 
-        // TODO: 1 in 5 projects use shared tables. Temporary until all projects are using shared tables.
-        if (
-            (
-                !\mt_rand(0, 4)
-                && System::getEnv('_APP_DATABASE_SHARED_TABLES', 'enabled') === 'enabled'
-                && System::getEnv('_APP_EDITION', 'self-hosted') !== 'self-hosted'
-            ) ||
-            (
-                $dsn === DATABASE_SHARED_TABLES
-            )
-        ) {
-            $schema = 'appwrite';
-            $database = 'appwrite';
-            $namespace = System::getEnv('_APP_DATABASE_SHARED_NAMESPACE', '');
-            $dsn = $schema . '://' . DATABASE_SHARED_TABLES . '?database=' . $database;
-
-            if (!empty($namespace)) {
-                $dsn .= '&namespace=' . $namespace;
-            }
-        }
-
-        // TODO: Allow overriding in development mode. Temporary until all projects are using shared tables.
-        if (
-            App::isDevelopment()
-            && System::getEnv('_APP_EDITION', 'self-hosted') !== 'self-hosted'
-            && $request->getHeader('x-appwrited-share-tables', false)
-        ) {
+        // TODO: Temporary until all projects are using shared tables.
+        if ($dsn === System::getEnv('_APP_DATABASE_SHARED_TABLES', '')) {
```

Comment:
does this mean it will either be shared table or non-shared table all together?

Reply:
It will still depend on which database is assigned, so if by random assignment or by a database override the user gets the shared tables db, that project will get shared tables.

For tests, since we only have one DB, it will be either shared tables or not. This is the main reason I added an extra test job that will run all tests for all service with shared tables. Actually gives us much better coverage as well since we know all existing tests pass with both methods.

Reply:
🆒